packages/logging/src/setup-logger-and-telemetry.ts

```diff
@@ -147,29 +147,33 @@ export function setupLoggerAndTelemetry(
     });
   });
 
-  bus.on('mongosh:start-session', function (args: SessionStartedEvent) {
-    const params = {
-      session_id,
-      userId,
-      telemetryAnonymousId,
-      ...args,
-    };
+  const toSnakeCase = (str: string): string => {
+    const matches = str.match(
+      /[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g
```

Comment:
Should we have some tests for this regex? It might also be nice to know what the expected input -> output might be in a comment, a bit hard to grasp at a glance.

Reply:
Yeah, the thing is that the API is also used in C++ in BoxedNode (with a different implementation, we do dynamic linking), so to ensure that we don't break anything because we are changing how BoxedNode works (for some reason), I still force to snake case the key before sending the event.

The snake casing is tested indirectly in packages/logging/src/setup-logger-and-telemetry.spec.ts, but I agree that it makes sense to have a specific tests for this.